avro2py/codegen.py

```diff
@@ -408,8 +409,29 @@ def visit_AnnAssign(self, node: ast.AnnAssign) -> ast.AnnAssign:
             return rewriter.visit(node)
 
 
+def resolve_schema_reference(schema: Record, schemas : List[Record]):
+    """Substitute the schema reference with the definition.""" 
+    if not schemas or schema.fully_qualified_name() in RESOLVED_SCHEMAS:
+        return
+
+    for field in schema.resolved_schema['fields']:
+        if isinstance(field['type'], str) and '.' in field['type']:
```

Comment:
If we're only looking at types that are strings, doesn't that mean we're going to fail to resolve references in nested/complex types (unions, arrays, maps, etc.), e.g. https://github.com/hchasestevens/avro2py/blob/master/example/schema.avsc#L1407-L1415 or https://github.com/hchasestevens/avro2py/blob/master/example/schema.avsc#L697-L700 ?